extensions/wikia/AdEngine/js/template/modal.js

```diff
@@ -18,8 +20,21 @@ define('ext.wikia.adEngine.template.modal', [
 	 * @param {number} params.height - desired height of the Lightbox
 	 */
 	function show(params) {
-		log(['showNew', params], 'debug', logGroup);
+		log(['show', params], 'debug', logGroup);
+		var skin = adContext.getContext().targeting.skin;
 
+		if (skin === 'oasis') {
+			log(['showNew desktop modal'], 'debug', logGroup);
+			createAndShowDesktopModal(params);
+		}
+
+		if (skin === 'mercury') {
+			log(['showNew mobile (Mercury) modal'], 'debug', logGroup);
+			win.Mercury.Modules.Ads.getInstance().openLightbox(createAdIframe(params));
```

Comment:
I don't think we want to have any graceful error handling here. If our tests don't catch it then somebody will sooner or later notice error in the console and we'll fix it :)